hail/src/main/scala/is/hail/expr/LowerArrayToStream.scala

```diff
@@ -0,0 +1,87 @@
+package is.hail.expr.ir
+
+import is.hail.expr.types.virtual.{TArray, TContainer, TStream, TStreamable}
+
+object LowerArrayToStream {
+  private def boundary(node: IR): IR = {
+    var streamified = streamify(node)
+
+    if (streamified.typ.isInstanceOf[TStream] && node.typ.isInstanceOf[TArray]) {
+      println("Wrapping")
+      streamified = ToArray(streamified)
+    }
+
+    if (streamified.typ.isInstanceOf[TArray] && node.typ.isInstanceOf[TStream])
```

Comment:
does this ever happen?

I think you only call boundary when you do not want a stream back, right?